app/services/sso/openid_connect.rb

```diff
@@ -0,0 +1,58 @@
+module SSO
+  class OpenidConnect < Base
+    delegate :session, :to => :controller
+    attr_reader :current_user
+
+    def available?
+      controller.api_request? && bearer_token_set? && valid_issuer?
+    end
+
+    def authenticate!
+      payload = jwt_token.decode
+      return nil if payload.nil?
+      user = find_or_create_user_from_jwt(payload)
+      @current_user = user
+      update_session(payload)
+      user&.login
+    end
+
+    def authenticated?
+      self.user = User.current.presence || authenticate!
+    end
+
+    private
+
+    def jwt_token
```

Comment:
I'd still like to avoid this naming (https://en.wikipedia.org/wiki/RAS_syndrome) but I won't block on it.

Reply:
I understand but we're already using that in multiple places and it seems pretty common in general (e.g. https://www.google.com/search?client=firefox-b-d&q=%22jwt+token%22).